src/shr_transport.h4

```diff
@@ -356,22 +356,22 @@ shmem_shr_transport_atomic(shmem_ctx_t ctx, void *target, const void *source,
 
 #define SHMEM_DEF_BAND_OP(STYPE,TYPE,ITYPE)                                                     \
                 case ITYPE:                                                                     \
-                    __atomic_fetch_and((TYPE *)remote_ptr, *((TYPE *)source), __ATOMIC_SEQ_CST);\
+                    __atomic_fetch_and((TYPE *)remote_ptr, *((TYPE *)source), __ATOMIC_RELEASE);\
```

Comment:
I would have guessed the `__atomic_fetch_[op]`'s would do release-acquire, because first there's a read, then a write.  Can you please explain the reasoning for release here (and below)?

Reply:
Maybe it's because we're not actually reading the return value...?

Reply:
That was my reasoning. We only have fetching atomic intrinsics, so I used those with release consistency to implement non-fetching OpenSHMEM AMOs.